client/tracing/src/lib.rs

```diff
@@ -37,12 +39,57 @@ use tracing::{
 	span::{Attributes, Id, Record},
 	subscriber::Subscriber,
 };
-use tracing_subscriber::{CurrentSpan, layer::{Layer, Context}};
+use tracing_subscriber::{
+	fmt::time::ChronoLocal,
+	CurrentSpan,
+	EnvFilter,
+	layer::{self, Layer, Context},
+	fmt as tracing_fmt,
+	Registry,
+};
 
 use sc_telemetry::{telemetry, SUBSTRATE_INFO};
 use sp_tracing::{WASM_NAME_KEY, WASM_TARGET_KEY, WASM_TRACE_IDENTIFIER};
+use tracing_subscriber::reload::Handle;
+use once_cell::sync::OnceCell;
+use tracing_subscriber::filter::Directive;
+
 const ZERO_DURATION: Duration = Duration::from_nanos(0);
 
+// The layered Subscriber as built up in `init_logger()`.
+// Used in the reload `Handle`.
+type SCSubscriber<
+	N = tracing_fmt::format::DefaultFields,
+	E = logging::EventFormat<ChronoLocal>,
+	W = fn() -> std::io::Stderr
+> = layer::Layered<tracing_fmt::Layer<Registry, N, E, W>, Registry>;
+
+static FILTER_RELOAD_HANDLE: OnceCell<Handle<EnvFilter, SCSubscriber>> = OnceCell::new();
+
+/// Initialize FILTER_RELOAD_HANDLE, only possible once
+pub fn set_reload_handle(handle: Handle<EnvFilter, SCSubscriber>) {
+	let _ = FILTER_RELOAD_HANDLE.set(handle);
+}
+
+/// Reload the logging filter with the supplied directives
+pub fn reload_filter(directives: String) -> Result<(), String> {
+	let mut env_filter = tracing_subscriber::EnvFilter::default();
```

Comment:
This means we oerride the default filters we always set in cli or not?

Reply:
This should be fixed.

Reply:
I think the only sane way to do this is to get rid of all current filters and then supply all new ones required; from the POV of user practicality this makes sense to me - also it is the way the underlying tracing system works - there is no way to do this piecemeal.

I think an RPC to get the current log directives could be useful, but would prefer as a follow-up PR

Reply:
Or we can have a raw_reload_log_filter and a reload_log_filter which uses the defaults. 

And this should be done in this pr.

Reply:
Ok that sounds good, I'll try it out...

Reply:
This should be resolved now